File: clients/algoliasearch-client-java-2/algoliasearch-core/com/algolia/model/search/Action.java
package com.algolia.model.search;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

/** type of operation. */
@JsonAdapter(Action.Adapter.class)
public enum Action {
  ADDOBJECT("addObject"),

  UPDATEOBJECT("updateObject"),

  PARTIALUPDATEOBJECT("partialUpdateObject"),

  PARTIALUPDATEOBJECTNOCREATE("partialUpdateObjectNoCreate"),

  DELETEOBJECT("deleteObject"),

  DELETE("delete"),

  CLEAR("clear");

  private String value;

  Action(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static Action fromValue(String value) {
    for (Action b : Action.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }

  public static class Adapter extends TypeAdapter<Action> {

    @Override
    public void write(final JsonWriter jsonWriter, final Action enumeration)
      throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public Action read(final JsonReader jsonReader) throws IOException {
      String value = jsonReader.nextString();
      return Action.fromValue(value);
    }
  }
}

File: clients/algoliasearch-client-php/lib/Model/Search/Action.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * Action Class Doc Comment
 *
 * @category Class
 * @description type of operation.
 *
 * @package Algolia\AlgoliaSearch
 */
class Action
{
    /**
     * Possible values of this enum
     */
    const ADD_OBJECT = 'addObject';

    const UPDATE_OBJECT = 'updateObject';

    const PARTIAL_UPDATE_OBJECT = 'partialUpdateObject';

    const PARTIAL_UPDATE_OBJECT_NO_CREATE = 'partialUpdateObjectNoCreate';

    const DELETE_OBJECT = 'deleteObject';

    const DELETE = 'delete';

    const CLEAR = 'clear';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [
            self::ADD_OBJECT,
            self::UPDATE_OBJECT,
            self::PARTIAL_UPDATE_OBJECT,
            self::PARTIAL_UPDATE_OBJECT_NO_CREATE,
            self::DELETE_OBJECT,
            self::DELETE,
            self::CLEAR,
        ];
    }
}

